### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by diegoh40**
- Set up associations between models.
- Write SQL queries with Active Record.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+  after_save :update_comment_counter
+
+  def comments_update
+    post.update(comments_counter: post.comments.all.length)
```
Comment (Deepakdanger) [inline app/models/comment.rb:7]: - [ ] Instead of using length kindly use `post.increment!(:comments_counter)`

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Like < ApplicationRecord
+  belongs_to :post
+  belongs_to :author, class_name: 'User'
+  after_save :update_like_counter
+
+  def likes_update
+    post.update(likes_counter: post.likes.all.length)
```
Comment (Deepakdanger) [inline app/models/like.rb:7]: - [ ] Instead of using length kindly use `post.increment!(:likes_counter)`

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  validates :title, presence: true, allow_blank: false, length: { maximum: 250 }
+
+  after_save :update_post_counter
+
+  def last_five_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def posts_update
+    author.update(posts_counter: author.posts.all.length)
```
Comment (Deepakdanger) [inline app/models/post.rb:15]: - [ ] Instead of using length kindly use  `author.increment!(:posts_counter)`

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @diegoh40 , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- 

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**